### ISSUE TITLE
Banana Pi M2 Zero with OctoPrint and camera

### ISSUE THREAD

**Issue by PurpleIsEverything** [mgx0/Banana-Pi-M2Z-Octoprint#1]
Thanks for this project! Have you done any testing with a CSI camera? With the extra compute power of the M2 Zero I'm wondering if it's more stable than the Pi Zero W.

**Comment by mgx0** [comment]
Camera wasn't tested, I didn't have the chance to buy it (raspberry camera does not fit banana), but from what I have seen this Banana Zero has really lots of power. Everything is much quicker than Raspberry Zero

**Comment by PurpleIsEverything** [comment]
Hmm, OK I might get one just to try it out if there is a good camera available for it. I want to install this all in my MK3S but I didn't really see the need for a full Pi 3 in a seperate case.

**Comment by BOFH90** [comment]
Offtopic: I also wanna install a Banana Pi Zero M2 in my mk3s. How are the Thermals of it as its known to overheat fairly quick. Thanks in advance.

**Comment by mgx0** [comment]
Hi, mine did not cause any issues during several 5-10hrs prints. I can only be subjective and say it does not need any cooling in the printer. Of course you can modify the cover and add a cutout for the heatsink

**Comment by BOFH90** [comment]
Thank you. I´ll guess i buy a BP Zero M2 and try your image :)

**Comment by mgx0** [comment]
Sure, no worries. Works fine for me, I hope it will also for you

**Comment by mgx0** [comment]
does the image work for you?

**Comment by BOFH90** [comment]
Still not got the chance to test as my BP Zero hasn´t arrived yet. Ordered it in China...

**Comment by Uko** [comment]
Wow, looks like the camera is a pain in the neck. The only discussions that I found about whether one can use a raspberry camera with a banana board are back from 2014 when there was even no official banana camera…

**Comment by mgx0** [comment]
yeah it is. raspberry pi camera cannot be used unfortunately. that's why I never got to the point where I tested it

**Comment by Uko** [comment]
@PurpleIsEverything @BOFH90 which cameras did you buy? Because the ones that I managed to find on [AliExpress](https://www.aliexpress.com/item/32660117929.html?spm=a2g0o.productlist.0.0.44f01d55u9N3eC&algo_pvid=40f50210-b728-4a9e-9150-162dd48e9a79&algo_expid=40f50210-b728-4a9e-9150-162dd48e9a79-0&btsid=94acd177-4671-4328-b98e-3270e028d70a&ws_ab_test=searchweb0_0,searchweb201602_9,searchweb201603_55) have a very short cable, and splicing ribbon cables is a pain…

**Comment by romainreignier** [comment]
With recent kernels (> 5.0), the camera interface on the H2 processor and the OV5640 sensor are supported.
So the only issue now is to find a longer cable for the camera...
